radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStateComputer.java

```diff
@@ -344,21 +358,19 @@ public StateComputerResult prepare(List<PreparedTxn> previous, VerifiedVertex ve
 			);
 		}
 
-		// Next epoch
-		if (proof.getNextValidatorSet().isPresent()) {
-			forks.ifForkGet(proof.getEpoch() + 1)
-				.ifPresent(rules -> {
-					log.info("Epoch {} Forking RadixEngine to {}", proof.getEpoch() + 1, rules.name());
-					this.radixEngine.replaceConstraintMachine(
-						rules.getConstraintMachineConfig(),
-						rules.getSerialization(),
-						rules.getActionConstructors(),
-						rules.getBatchVerifier(),
-						rules.getParser()
-					);
-					this.epochCeilingView = rules.getMaxRounds();
-				});
-		}
+		maybeNextForkConfig.ifPresent(nextForkConfig -> {
```

Comment:
This should really be done only on epoch changes as it is not guaranteed everyone will receive proof of some txn in between an epoch.

Reply:
done